[API/YAML changes] Scope Fleet-maintained apps and custom packages via labels (#24085)
fleetdm/fleet: docs/REST API/rest-api.md

| pre_install_query  | string | form | Query that is pre-install condition. If the query doesn't return any result, the package will not be installed. |
| post_install_script | string | form | The contents of the script to run after install. If the specified script fails (exit code non-zero) software install will be marked as failed and rolled back. |
| self_service | boolean | form | Whether this is optional self-service software that can be installed by the end user. |

> | labels_include_any        | array     | form | Software will only be available for install or installed (by policy automation) on hosts that **have any** of these labels. Only one of either `labels_include_any` or `labels_exclude_any` can be included in the request. |
> | labels_exclude_any | array | form | Software will only be available for install or installed (by policy automation) on hosts that **don't have any** of these labels. Only one of either `labels_include_any` or `labels_exclude_any` can be included in the request. |

Reviewer: (See comment below re: modify endpoints) 

```suggestion
| labels_include_any        | array     | form | Software will only be available for install or installed (by policy automation) on hosts that **have any** of these labels. Only one of either `labels_include_any` or `labels_exclude_any` can be specified. |
| labels_exclude_any | array | form | Software will only be available for install or installed (by policy automation) on hosts that **don't have any** of these labels. Only one of either `labels_include_any` or `labels_exclude_any` can be specified. |
```

`Status: 200`

> ##### Default response
> 
> `Status: 200`

Reviewer: Looks like this is a duplicate heading/status
```suggestion
```

`Status: 200`

### Modify App Store app

> **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows.

_Available in Fleet Premium._

Modify App Store (VPP) app's options.

`PATCH /api/v1/fleet/software/titles/:title_id/app_store_app`

#### Parameters

| Name | Type | In | Description |
| ---- | ---- | -- | ----------- |
| team_id       | integer | body | **Required**. The team ID. Edits App Store apps from the specified team.  |
| self_service | boolean | body | Self-service software is optional and can be installed by the end user. |

> | labels_include_any        | array     | form | App will only be available for install or installed (by policy automation) on hosts that **have any** of these labels. Only one of either `labels_include_any` or `labels_exclude_any` can be included in the request. |
> | labels_exclude_any | array | form | App will only be available for install or installed (by policy automation) on hosts that **don't have any** of these labels. Only one of either `labels_include_any` or `labels_exclude_any` can be included in the request. |

Reviewer: I think we need to reword this for PATCH requests; what happens if `labels_exclude_any` was previously set, and this API request has `labels_include_any`? (I'm assuming it throws an error?)

```suggestion
| labels_include_any        | array     | form | App will only be available for install or installed (by policy automation) on hosts that **have any** of these labels. Only one of either `labels_include_any` or `labels_exclude_any` can be specified. |
| labels_exclude_any | array | form | App will only be available for install or installed (by policy automation) on hosts that **don't have any** of these labels. Only one of either `labels_include_any` or `labels_exclude_any` can be specified. |
```

#### Example

`PATCH /api/v1/fleet/software/titles/3467/app_store_app`

##### Request body

```json
{
  "team_id": 2,
  "self_service": true,
  "labels_include_any": [
    "Product",
    "Marketing"
  ]
}
```

##### Default response

> `Status: 200`

Reviewer: For `PATCH` endpoints, we usually return the modified record as well ([example](https://fleetdm.com/docs/rest-api/rest-api#default-response125))
